### ISSUE TITLE
lcobucci/jwt don't include directly in auth0/login. It should be externally. (Warning: Ambiguous class resolution)

### ISSUE THREAD

**Issue by rajnarayan-pix** [auth0/laravel-auth0#235]
### SDK Version

6.4

### PHP Version

PHP 8.0

### Composer Version

2.x

### What happened?

lcobucci/jwt don't include directly in auth0/login. It should be externally. It helps resolving issue.

### How can we reproduce this issue?

Laravel Version : 8.40
PHP Version : 8
Passport Version : 10.2
auth0/login : 6.4.0

Using passport and auth0/login in laravel 8 project where when I run command composer install getting below warning.

`Warning: Ambiguous class resolution, "Lcobucci\JWT\Builder" was found 2x: in "/var/www/vendor/auth0/php-jwt/src/Builder.php" and "/var/www/vendor/lcobucci/jwt/src/Builder.php", the first will be used.
Warning: Ambiguous class resolution, "Lcobucci\JWT\Parser" was found 2x: in "/var/www/vendor/auth0/php-jwt/src/Parser.php" and "/var/www/vendor/lcobucci/jwt/src/Parser.php", the first will be used.
Warning: Ambiguous class resolution, "Lcobucci\JWT\Signer\Ecdsa\MultibyteStringConverter" was found 2x: in "/var/www/vendor/auth0/php-jwt/src/Signer/Ecdsa/MultibyteStringConverter.php" and "/var/www/vendor/lcobucci/jwt/src/Signer/Ecdsa/MultibyteStringConverter.php", the first will be used.
Warning: Ambiguous class resolution, "Lcobucci\JWT\Signer\Ecdsa\Sha256" was found 2x: in "/var/www/vendor/auth0/php-jwt/src/Signer/Ecdsa/Sha256.php" and "/var/www/vendor/lcobucci/jwt/src/Signer/Ecdsa/Sha256.php", the first will be used.
Warning: Ambiguous class resolution, "Lcobucci\JWT\Signer\Ecdsa\Sha384" was found 2x: in "/var/www/vendor/auth0/php-jwt/src/Signer/Ecdsa/Sha384.php" and "/var/www/vendor/lcobucci/jwt/src/Signer/Ecdsa/Sha384.php", the first will be used.
Warning: Ambiguous class resolution, "Lcobucci\JWT\Signer\Ecdsa\Sha512" was found 2x: in "/var/www/vendor/auth0/php-jwt/src/Signer/Ecdsa/Sha512.php" and "/var/www/vendor/lcobucci/jwt/src/Signer/Ecdsa/Sha512.php", the first will be used.
Warning: Ambiguous class resolution, "Lcobucci\JWT\Signer\Ecdsa\SignatureConverter" was found 2x: in "/var/www/vendor/auth0/php-jwt/src/Signer/Ecdsa/SignatureConverter.php" and "/var/www/vendor/lcobucci/jwt/src/Signer/Ecdsa/SignatureConverter.php", the first will be used.
Warning: Ambiguous class resolution, "Lcobucci\JWT\Signer\Ecdsa" was found 2x: in "/var/www/vendor/auth0/php-jwt/src/Signer/Ecdsa.php" and "/var/www/vendor/lcobucci/jwt/src/Signer/Ecdsa.php", the first will be used.
Warning: Ambiguous class resolution, "Lcobucci\JWT\Signer\Hmac\Sha256" was found 2x: in "/var/www/vendor/auth0/php-jwt/src/Signer/Hmac/Sha256.php" and "/var/www/vendor/lcobucci/jwt/src/Signer/Hmac/Sha256.php", the first will be used.
Warning: Ambiguous class resolution, "Lcobucci\JWT\Signer\Hmac\Sha384" was found 2x: in "/var/www/vendor/auth0/php-jwt/src/Signer/Hmac/Sha384.php" and "/var/www/vendor/lcobucci/jwt/src/Signer/Hmac/Sha384.php", the first will be used.
Warning: Ambiguous class resolution, "Lcobucci\JWT\Signer\Hmac\Sha512" was found 2x: in "/var/www/vendor/auth0/php-jwt/src/Signer/Hmac/Sha512.php" and "/var/www/vendor/lcobucci/jwt/src/Signer/Hmac/Sha512.php", the first will be used.
Warning: Ambiguous class resolution, "Lcobucci\JWT\Signer\Hmac" was found 2x: in "/var/www/vendor/auth0/php-jwt/src/Signer/Hmac.php" and "/var/www/vendor/lcobucci/jwt/src/Signer/Hmac.php", the first will be used.
Warning: Ambiguous class resolution, "Lcobucci\JWT\Signer\Key" was found 2x: in "/var/www/vendor/auth0/php-jwt/src/Signer/Key.php" and "/var/www/vendor/lcobucci/jwt/src/Signer/Key.php", the first will be used.
Warning: Ambiguous class resolution, "Lcobucci\JWT\Signer\OpenSSL" was found 2x: in "/var/www/vendor/auth0/php-jwt/src/Signer/OpenSSL.php" and "/var/www/vendor/lcobucci/jwt/src/Signer/OpenSSL.php", the first will be used.
Warning: Ambiguous class resolution, "Lcobucci\JWT\Signer\Rsa\Sha256" was found 2x: in "/var/www/vendor/auth0/php-jwt/src/Signer/Rsa/Sha256.php" and "/var/www/vendor/lcobucci/jwt/src/Signer/Rsa/Sha256.php", the first will be used.
Warning: Ambiguous class resolution, "Lcobucci\JWT\Signer\Rsa\Sha384" was found 2x: in "/var/www/vendor/auth0/php-jwt/src/Signer/Rsa/Sha384.php" and "/var/www/vendor/lcobucci/jwt/src/Signer/Rsa/Sha384.php", the first will be used.
Warning: Ambiguous class resolution, "Lcobucci\JWT\Signer\Rsa\Sha512" was found 2x: in "/var/www/vendor/auth0/php-jwt/src/Signer/Rsa/Sha512.php" and "/var/www/vendor/lcobucci/jwt/src/Signer/Rsa/Sha512.php", the first will be used.
Warning: Ambiguous class resolution, "Lcobucci\JWT\Signer\Rsa" was found 2x: in "/var/www/vendor/auth0/php-jwt/src/Signer/Rsa.php" and "/var/www/vendor/lcobucci/jwt/src/Signer/Rsa.php", the first will be used.
Warning: Ambiguous class resolution, "Lcobucci\JWT\Signer" was found 2x: in "/var/www/vendor/auth0/php-jwt/src/Signer.php" and "/var/www/vendor/lcobucci/jwt/src/Signer.php", the first will be used.
Warning: Ambiguous class resolution, "Lcobucci\JWT\Token" was found 2x: in "/var/www/vendor/auth0/php-jwt/src/Token.php" and "/var/www/vendor/lcobucci/jwt/src/Token.php", the first will be used.`

And when trying to generate access token getting below error
`{
  "message": "Lcobucci\\JWT\\Signer\\Key\\InMemory cannot implement Lcobucci\\JWT\\Signer\\Key - it is not an interface",
  "context": {
    "exception": {
      "class": "Symfony\\Component\\ErrorHandler\\Error\\FatalError",
      "message": "Lcobucci\\JWT\\Signer\\Key\\InMemory cannot implement Lcobucci\\JWT\\Signer\\Key - it is not an interface",
      "code": 0,
      "file": "/var/www/vendor/lcobucci/jwt/src/Signer/Key/InMemory.php:14"
    }
  },
  "level": 400,
  "level_name": "ERROR",
  "channel": "local",
  "datetime": "2021-11-02T06:20:12.672273+00:00",
  "extra": {}
}`

**Comment by bm-srdjanmarjanovic** [comment]
I had this same problem

**Comment by evansims** [comment]
Hi @rajnarayan-pix 👋 Sorry for the frustration; this is a known issue at present.

The current release of our Laravel plugin uses a forked version of the mentioned library that allows us to shim in PHP 8 support. The library's author did not want to backport PHP 8 support into the version of the dependency we relied on, which required us to build a workaround in the meantime.

This dependency conflict was introduced in the upstream auth0-PHP v7, which this library relies on, to allow us to support PHP 8.0. The problematic dependency was removed in auth0-PHP v8, and we're in the process of updating this library to use that SDK version instead, which will resolve this issue. We're hoping to have a new version of this library out yet this quarter that resolves this issue.
